screens/MoreDetails.js

```diff
@@ -1,12 +1,278 @@
 import React from 'react';
-import {View, Text} from 'react-native'
+import { View, Text, StyleSheet, Image } from 'react-native'
+import { Icon, Button, Right, Separator } from 'native-base';
+import { sgwData } from '../constants/sgwData';
+import { ScrollView } from 'react-native-gesture-handler';
 
-function MoreDetails (props) {
-    return(
-        <View>
-            <Text>{props.name}</Text>
-        </View>  
+
+function MoreDetails(props) {
+
+
+
+    const getBuildingInfo = sgwData();
+    const dept = getBuildingInfo[props.name].departments;
+    const services = getBuildingInfo[props.name].services;
+
+
+
+    return (
```

Comment:
I would have personally put everything in a _buildingInfo_ component. So in that return you would only write that component. That would allow us to have other components like _PointOfInterestComponent_ that will allow us to make the layout for a restaurant or something. Because a potential problem that I see with writting all layout logic in _moredetails_ component is that, it will be very long and tedious in terms of logic.

Reply:
Good point! It will truly help for the next step!